examples/zonal_private_iap_bastion/outputs.tf

```diff
@@ -0,0 +1,132 @@
+}
+
+output "region" {
+  description = "Cluster region"
+  value       = "${module.gke.region}"
+}
+
+output "zones" {
+  description = "List of zones in which the cluster resides"
+  value       = "${module.gke.zones}"
+}
+
+output "endpoint" {
+  sensitive   = true
+  description = "Cluster endpoint"
+  value       = "${module.gke.endpoint}"
+}
+
+output "min_master_version" {
+  description = "Minimum master kubernetes version"
+  value       = "${module.gke.min_master_version}"
+}
+
+output "logging_service" {
+  description = "Logging service used"
+  value       = "${module.gke.logging_service}"
+}
+
+output "monitoring_service" {
+  description = "Monitoring service used"
+  value       = "${module.gke.monitoring_service}"
+}
+
+output "master_authorized_networks_config" {
+  description = "Networks from which access to master is permitted"
+  value       = "${module.gke.master_authorized_networks_config}"
+}
+
+output "master_version" {
+  description = "Current master kubernetes version"
+  value       = "${module.gke.master_version}"
+}
+
+output "ca_certificate" {
+  sensitive   = true
+  description = "Cluster ca certificate (base64 encoded)"
+  value       = "${module.gke.ca_certificate}"
+}
+
+output "network_policy_enabled" {
+  description = "Whether network policy enabled"
+  value       = "${module.gke.network_policy_enabled}"
+}
+
+output "http_load_balancing_enabled" {
+  description = "Whether http load balancing enabled"
+  value       = "${module.gke.http_load_balancing_enabled}"
+}
+
+output "horizontal_pod_autoscaling_enabled" {
+  description = "Whether horizontal pod autoscaling enabled"
+  value       = "${module.gke.horizontal_pod_autoscaling_enabled}"
+}
+
+output "kubernetes_dashboard_enabled" {
+  description = "Whether kubernetes dashboard enabled"
+  value       = "${module.gke.kubernetes_dashboard_enabled}"
+}
+
+output "node_pools_names" {
+  description = "List of node pools names"
+  value       = "${module.gke.node_pools_names}"
+}
+
+output "node_pools_versions" {
+  description = "List of node pools versions"
+  value       = "${module.gke.node_pools_versions}"
+}
+
+output "service_account" {
+  description = "The service account to default running nodes as if not overridden in `node_pools`."
+  value       = "${module.gke.service_account}"
+}
+
+output "network_name" {
+  value       = "${module.gke-network.network_name}"
+  description = "The name of the VPC being created"
+}
+
+output "get_credentials" {
+  description = "Gcloud get-credentials command"
+  value       = format("gcloud container clusters get-credentials --project %s --zone %s --internal-ip %s", var.project_id, module.gke.location, var.cluster_name)
+}
+
+output "bastion_ssh" {
+  description = "Gcloud compute ssh to the bastion host command"
+  value       = format("gcloud beta compute ssh %s --tunnel-through-iap --project %s --zone %s -- -L8888:127.0.0.1:8888", module.bastion.hostname, var.project_id, local.bastion_zone)
```

Comment:
Tunnel shouldn't be to 127.0.0.1:8888, which would connect to the bastion host.  It should be to the private endpoint of the gke cluster

-L8888:${module.gke.endpoint}:8888 or -L8888:%s:8888 and add module.gke.endpoint to the format parameter list

That seems to fix it, but I have to fix my firewall to allow connections on port 8888, first, I think.  Mine is just hanging when trying to connect across that tunnel - but it used to error out with the tunnel to 127.0.0.1:8888

Reply:
Bastion is running TinyProxy, so the IAP tunnel is allowing us to proxy 8888 from localhost through the bastion and finally to the cluster master, which allows us to run kubectl commands through. What was the error when it was set to 127.0.0.1:8888?

Reply:
Failure to connect. 

But I’ll check it again when I get back to work on it later. I have tiny proxy installed on mine, too, but i didn’t dig into the logs and make sure the startup script ran correctly and the proxy started up. I didn’t grok that tinyproxy was intended to be receiving that connection, so I didn’t even try to debug the failed connection. I just switched it to the endpoint address.

I’m curious, what’s the purpose of using tiny proxy instead of setting up the tunnel to connect directly to the endpoint? Just the convenience of being able to connect to any internal address and port from that one tunnel, instead of one tunnel for each service being connected to? Or is there more to it?

Reply:
Confirmed - tinyproxy didn’t install due to lack of outbound internet access. User error

Reply:
The intent was to lock down access to the cluster to be all private. The firewall is only configured to allow certain connections in from the bastion host, and only resources inside the VPC are configured in the master authorized network list for the cluster.

Reply:
I understand that, but if the bastion host can get to the k8s cluster, then `-L8888:${private_endpoint}:8888` works in the exact same way, just without the extra hop through a proxy server on the bastion.  In either case, port 8888 on the ssh client host connects to the bastion and then connects to the specified destination address on the destination port from there.  Any DNS lookup, if a hostname is used instead of ip address, also occurs on the bastion, not on the ssh client host, so you can use internal DNS names there.

Reply:
I ran into a similar issue when I was migrating the system over to use the bastion host module. TinyProxy did not install due to the image being set to centos instead of debian, and I did not realize the startup script did not run. I agree that documentation is a good idea, especially for those who cannot pull down tiny proxy. I will add it and resolve this conversation once I make the updates.

Reply:
Success! I finally got everything to work.  I have the fabric firewall module in play (with defaults, so traffic is fairly well locked down), and I had to muck around with firewall rules and routes a bit, but everything is now functional.  

This example was super helpful, too.  It probably would have taken me several more days to get it all put together.

Changes I made to make it work on a network with default firewall rules:
- I had to make sure the subnet my bastion is in was added to the authorized networks, which also sets up the route
- I added a firewall rule which allows the bastion ip range to target all ports and protocols in the gke service account, but I suspect I can also make it use bastion source tag and gke target tag - I just haven't tested it yet.  The second form is more flexible.

These are the changes I made to get the bastion to startup with the proxy correctly, but a network configured differently would probably get access to the public internet differently:
- I had to add tags to the bastion host (PR already merged to support that), in order to create a route for the bastion out to the public internet, which fixed the proxy install (in combination with a firewall rule that allows the traffic).
- I added a firewall rule to allow outbound https traffic to 0.0.0.0/0 from a bastion tag (for fetching packages)

I think everything else just worked as-is when copied from your example.

Reply:
@wripley I wonder if it would be helpful if we included explicit firewall rules in this example? Even though we're spinning up a new network (which will by default allow all egress), it might be helpful to show what you need to allow in a more locked down network.

Reply:
We could show an egress firewall rule being allowed for the bastion host.

Reply:
I haven’t terraformed the routes and rules I created manually to get it all working yet, but I will do so in the next 24 hours. I’ll paste something here once I’ve got them - save you most of the work.

Reply:
Perhaps consider adding `-A` to the ssh args, which enables ssh agent forwarding, in order to be able to bounce through the bastion to node_pool instances.